lib/widgets/resources/arm_widgets/pose_widget.dart

```diff
@@ -30,16 +35,44 @@ class _PoseWidgetState extends State<PoseWidget> {
     _getStartPose();
   }
 
+  Pose controlValuesToPose() {
+    return Pose(
+      x: _controlValues[0],
+      y: _controlValues[1],
+      z: _controlValues[2],
+      oX: _controlValues[3],
+      oY: _controlValues[4],
+      oZ: _controlValues[5],
+      theta: _controlValues[6],
+    );
+  }
+
   Future<void> _getStartPose() async {
     final startPose = await widget.arm.endPosition();
-    _controlValues = [startPose.x, startPose.y, startPose.z, startPose.oX, startPose.oY, startPose.oZ, startPose.theta];
+    _controlValues = [
+      startPose.x.roundToDouble(),
+      startPose.y.roundToDouble(),
+      startPose.z.roundToDouble(),
+      startPose.oX.roundToDouble(),
+      startPose.oY.roundToDouble(),
+      startPose.oZ.roundToDouble(),
+      startPose.theta.roundToDouble(),
+    ];
```

Comment:
missed this in the last PR (sorry!), but representing the `_controlValues` as a list seems not great to me. Very easy to run into issues if at any point the data representation changes. Would much prefer to have some sort of struct with named fields to make sure that we're always setting the right data to the right place.